### DISCUSSION TITLE
ec2-instance-connect and open-tunnel

### DISCUSSION THREAD

**Discussion by object88** [awslabs/aws-sdk-rust#1057] [Q&A] [answered]
I'm trying my first "real" Rust project; I want to be able to SSH into a private EC2 instance using EC2 Instance Connect.

I have most of the first "dots" connected; I am able to query AWS to get the instance's subnet, az, etc.  I am able to generate an SSH keypair, and send the public key to AWS, apparently successfully.

In the [fine documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/connect-using-eice.html#eic-connect-using-ssh), it has an example of invoking openssh:
```
ssh -i my-key-pair.pem ec2-user@i-0123456789example \
    -o ProxyCommand='aws ec2-instance-connect open-tunnel --instance-id i-0123456789example'
```
The `open-tunnel` functionality does not appear to be implemented in the Rust SDK (nor in the Go SDK, IIRC).  Reading some of the CLI [code](https://github.com/hsingh6764/aws-cli/blob/2.13.1/awscli/customizations/ec2instanceconnect/opentunnel.py#L142-L146) (and [here](https://github.com/hsingh6764/aws-cli/blob/2.13.1/awscli/customizations/ec2instanceconnect/websocket.py#L290)), it _feels_ like it's essentially using a websocket..., something I have no practical experience with.

Is this something that may be coming in a future revision of the Rust SDK?  Is this a "no-brainer", and if so, can someone point me to a reasonable websockets library that I can use?  I figure that I can use the above code to sort out the endpoints, etc -- assuming that those are not considered "internals" and subject to change?

Thanks for any help && pointers!

**Comment by tdalbo92** [comment]
Yeah, I have aws cli installed via python and it's the same thing.  `open-tunnel` does not exist on my version of aws-cli.

**Reply by jmklix** [reply, marked as answer]
Sorry, but this is a cli customization that we currently don't have any plans to add to the rust sdk
